flatfs.go

```diff
@@ -404,6 +587,127 @@ func (fs *Datastore) walkTopLevel(path string, reschan chan query.Result) error
 	return nil
 }
 
+// calculateDiskUsage tries to read the DiskUsageFile for a cached
+// diskUsage value, otherwise walks the datastore files.
+func (fs *Datastore) calculateDiskUsage() error {
+	// Try to obtain a previously stored value from disk
+	if persDu := fs.readDiskUsageFile(); persDu > 0 {
+		atomic.StoreInt64(&fs.diskUsage, persDu)
+		return nil
+	}
+
+	// Otherwise walk the datastore files
+	var du int64
+	err := filepath.Walk(fs.path,
```

Comment:
I would not use filepath.Walk here.  It sorts the directory names which is unnecessary and might lead to excessive seeking as stat is called on each directory entry.  It could also be doing other unnecessary things.  There is no reason this can't as efferent as `du`.  I would use `Readdir` directly.